code/modules/fabrication/designs/_design.dm

```diff
@@ -57,14 +57,26 @@
 
 /obj/building_cost()
 	. = ..()
+	if(!simulated || INSTANCE_IS_ABSTRACT(src))
+		return
 	if(length(matter))
+		LAZYINITLIST(.)
 		for(var/material in matter)
 			var/decl/material/M = GET_DECL(material)
 			if(istype(M))
-				.[M.type] = matter[material]
+				.[M.type] += matter[material]
 	if(reagents && length(reagents.reagent_volumes))
+		LAZYINITLIST(.)
 		for(var/R in reagents.reagent_volumes)
-			.[R] = FLOOR(REAGENT_VOLUME(reagents, R) / REAGENT_UNITS_PER_MATERIAL_UNIT)
+			.[R] += FLOOR(REAGENT_VOLUME(reagents, R) / REAGENT_UNITS_PER_MATERIAL_UNIT)
+	for(var/obj/thing in contents)
```

Comment:
Should this be `contents` or `get_contained_external_atoms()`? Note that this issue is why I made `get_contained_matter` for calculating this type of thing at runtime: it has some overlap with the objects in `get_contained_external_atoms` but adds/removes extra ones, like machine parts, etc.